inc/Engine/Admin/Settings/Subscriber.php

```diff
@@ -73,7 +73,21 @@ public static function get_subscribed_events() {
 			'admin_notices'                        => 'display_update_notice',
 		];
 
-		return array_merge( $events, PluginFamily::get_subscribed_events() );
+		foreach ( PluginFamily::get_subscribed_events() as $hook => $callback ) {
+			if ( isset( $events[ $hook ] ) ) {
+				// Make sure it's an array of callbacks.
+				if ( ! is_array( $events[ $hook ][0] ) ) {
+					$events[ $hook ] = [ $events[ $hook ] ];
+				}
+
+				// Wrap single callback in array if needed.
+				$events[ $hook ][] = is_array( $callback ) ? $callback : [ $callback ];
+			} else {
+				$events[ $hook ] = is_array( $callback ) ? $callback : [ [ $callback ] ];
+			}
+		}
+
+		return $events;
```

Comment:
```suggestion
		foreach ( PluginFamily::get_subscribed_events() as $hook => $callback ) {
			if ( ! isset( $events[ $hook ] ) ) {
				continue;
			}
			$events[ $hook ] = array_merge( (array) $events[ $hook ], (array) $callback );
		}
```
What do u think about this?

Reply:
@wordpressfan 

Thanks for the suggestion! Your version is definitely shorter, but I think it might cause a couple of issues:

-   It skips any hooks that only exist in `PluginFamily::get_subscribed_events()`. For example, if PluginFamily defines `'admin_notices' => [ 'display_error_notice', 10 ]` and that hook doesn’t already exist in `$events`, it just gets ignored and never registered.
    
-   It can also introduce nested arrays if the callbacks are already structured as [ 'callback', 10, 1 ]. For example:
``` php
$events[ $hook ] = array_merge( [ $events[ $hook ] ], [ $callback ] );
```
could result in:
``` php
[
  [ 'existing_callback', 10, 2 ],
  [ 'new_callback' ]
]
```
but if $events[ $hook ] was already an array of arrays, this ends up being:
``` php
[
  [ [ 'existing_callback', 10, 2 ] ],
  [ 'new_callback' ]
]
```

-   which breaks the expected format.
    

The current version is more verbose, but it handles:

-   callbacks defined as strings (`'my_callback'`)
    
-   arrays (`[ 'my_callback' ]`)
    
-   and full definitions with priority and args (`[ 'my_callback', 10, 2 ]`)
   

…while also making sure new hooks from PluginFamily are added.

  

Open to tweaking it further if you see a cleaner way that handles all those cases!